Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,41 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("given an array with one number, returns that number", () => {
+    expect(sum([5])).toBe(5);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("given an array with negative numbers, returns correct sum", () => {
+    expect(sum([1, -2, 3, -4])).toBe(-2);
+});
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("given an array with decimal numbers, returns correct sum", () => {
+    expect(sum([1.5, 2.7, 3.2])).toBe(7.4);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?